### ISSUE TITLE
Example for models with modifiers

### ISSUE THREAD

**Issue by qchar** [alex-w0/objection-js-soft-delete#56]
Could you provide an example of using this plugin with models that are using modifiers? [DOC](https://vincit.github.io/objection.js/recipes/modifiers.html). Or is this impossible?

Currently, after migrating from `objection-soft-delete` to `objection-js-soft-delete` I'm getting errors like:

```
Cannot set property modifiers of class extends Model {
  static get QueryBuilder() {
    return SDQueryBuilder;
  }
  static get...<omitted>... } which has only a getter
```

I'm defining my models like:

```
export class Event extends softDelete({
  columnName: "archivedAt",
  deletedValue: database.fn.now() as unknown as number, // this type also was problematic after migration
  notDeletedValue: null,
})(Model) {
  
  [...]

  static modifiers =  {
    // my custom modifiers
  }
}

```

I saw #55 and #36, but I don't know if it is relevant in my case.

**Comment by alex-w0** [comment]
No, that should be totally possible. I've tested it and could not find a problem with the following code.

```
class Event extends softDelete(options)(Model) {
      
      static modifiers = {
          ...super.modifiers,
          customModifier(query) {
              query.select('id', 'firstName');
          },
      };
  }
```

Maybe you are not using an [ECMAScript bundle loader](https://www.npmjs.com/package/esm) and so your environment can not understand the syntax?

It's quite difficult to get the correct type for database.fn.now(), because it's primary used as a knex function helper. Maybe use `new Date()` or do you have a better idea for the typing? Or should we set the type to unknown and I don't mind anymore which type it is?

**Comment by qchar** [comment]
Thanks for the quick answer. It helped me to kind-of-resolve the problem with custom modifiers. 

When I tried the code above, I've got another TS error: `'super' property access is permitted only in a constructor, member function, or member accessor of a derived class.ts(2338)`. And then, I realized that you are using different syntax in your source code when defining `notDeleted` and `deleted` modifiers. This is because you are [defining](https://github.com/alex-w0/objection-js-soft-delete/blob/master/src/softDelete.js#L65) a function, not an object. So I went back to objection.js documentation and found out that in API reference, they are also using [functions](https://vincit.github.io/objection.js/api/model/static-properties.html#static-modifiers), differently than in [recipies](https://vincit.github.io/objection.js/recipes/modifiers.html).

So to summarize, this works for me:

```
static get modifiers() {
    return {
      ...super.modifiers,
      // my custom modifiers
    };
  }
```

and this doesn't:
```
static modifiers = {
  // my custom modifiers
}
```

If both syntaxes should work(I'm using node 14.x), maybe there is something wrong with my tsconfig.json:

```
{
  "compilerOptions": {
    "target": "es2019",
    "module": "commonjs",
    "lib": ["es2019"],
    "sourceMap": true,
    "outDir": "./dist",
    "strict": true,
    "moduleResolution": "node",
    "esModuleInterop": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "skipLibCheck": true,
    "forceConsistentCasingInFileNames": true,
  },
  "include": ["src/**/*"]
}
```

As far as this type of problem with `knex.fn.now()` goes, I had some issues with timezones when using `new Date()`. That's why I stick with this 'DB-native' syntax. So maybe this `deletedValue` type should be an `unknown`. In knex types this is basically defined as `any` type, or to be precise `Raw<any>`:

```
  interface FunctionHelper {
    now(precision?: number): Raw;
  }
```

Thanks once more for your work on this library.
